### PR TITLE
Fix gas calculation

### PR DESCRIPTION
**Pull request by piotmag769**
## Introduced changes

- now gas used for constructor execution and l1 handler execution is included in total gas cost
- `DeployPayload` and `CallOutput` struct were removed (no longer necessary)

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs [hunk 1]
```diff
@@ -228,15 +220,16 @@ pub fn call_entry_point(
...
+    let call_result = CallResult::from_execution_result(&exec_result, starknet_identifier);
+
+    let used_resources = UsedResources {
+        execution_resources: resources,
+        l2_to_l1_payloads_length: exec_result.map_or(vec![], |call_info| {
+            call_info.get_sorted_l2_to_l1_payloads_length().unwrap()
+        }),
+    };
+    // add execution resources used by call contract, library call or l1 handler execution to all used resources
+    cheatnet_state.used_resources.extend(&used_resources);
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs:232]: important change

File: crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/deploy.rs [hunk 2]
```diff
@@ -91,6 +82,13 @@ pub fn deploy_at(
         CheatcodeError::from(call_contract_failure)
     });
     cheatnet_state.increment_deploy_salt_base();
+
+    // add execution resources used by deploy to all used resources
+    cheatnet_state.used_resources.extend(&UsedResources {
+        execution_resources: resources,
+        l2_to_l1_payloads_length: vec![],
+    });
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/deploy.rs:90]: important change

**Review by piotmag769 — COMMENTED** [review]
Only changes marked as important + tests are relevant, the rest are signature and usages changes

**Review by drknzz — APPROVED** [review]
lgtm, our cheatcodes don't include syscalls like deploy / call contract in the resources but i guess it's out of scope for this one

**Review by Arcticae — APPROVED** [review]

**Pull request merged**